lib/board.rb

```diff
@@ -1,33 +1,96 @@
 class Board
 
-  def create_grid(dimension)
-    size = Math.sqrt(dimension)
+  attr_reader :dimension, :grid
+
+  def initialize(dimension)
+    @dimension = dimension
+    @grid = create_grid
+  end
+
+  def create_grid
+    possible_rows.inject({}) { |result, element| result[element] = possible_columns; result }
```

Comment:
If you need the semi colon then this shouldn't be on one line any more.  Swap to a do/end version.